### PR TITLE
CLOUD-616 run linters automatically on each PR

### PR DESCRIPTION
**Pull request by delgod**
[CLOUD-616](https://jira.percona.com/browse/CLOUD-616) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

check code automatically with the following linters:
 * golangci-lint
 * go-consistent
 * gofmt
 * shfmt
 * shellcheck
 * misspell
 * alex (insensitive, inconsiderate wording)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| recreate | passed |
| haproxy | passed |
| storage | passed |
| scaling-proxysql | failed |
| init-deploy | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| security-context | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| smart-update | passed |
| operator-self-healing | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/791/commits/9e6bfed30dd24acbe2e039c3a606741363696f56
image: `perconalab/percona-xtradb-cluster-operator:PR-791-9e6bfed3`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | failed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| storage | passed |
| self-healing | passed |
| users | passed |
| self-healing-advanced | passed |
| tls-issue-self | passed |
| operator-self-healing | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| big-data | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| security-context | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/791/commits/9e6bfed30dd24acbe2e039c3a606741363696f56
image: `perconalab/percona-xtradb-cluster-operator:PR-791-9e6bfed3`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| big-data | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| storage | passed |
| one-pod | passed |
| self-healing | passed |
| auto-tuning | passed |
| self-healing-advanced | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| scaling | passed |
| users | passed |
| scaling-proxysql | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| security-context | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/791/commits/9e6bfed30dd24acbe2e039c3a606741363696f56
image: `perconalab/percona-xtradb-cluster-operator:PR-791-9e6bfed3`

**Review by qJkee — APPROVED** [review]
i think we need more checks

File: .github/workflows/reviewdog.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          cd .github/linters
+          go get github.com/quasilyte/go-consistent
+
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          reviewdog -conf=./.reviewdog.yml -reporter=github-pr-check -runners=go-consistent -fail-on-error
+
+  gofmt:
```
Comment (blez) [inline .github/workflows/reviewdog.yml:40]: As  I know `gofmt` is a part of `golangci-lint` so maybe we could just turn it on there.

**Review by blez — APPROVED** [review]

**Pull request merged**